### PR TITLE
test: Add Tests for Reposting Entries, Posting Time Conversion, and Data Validation in Stock and Account Value Comparison Report

### PR DESCRIPTION
**Pull request by ashishverma8848**
### Title: Comprehensive Test Case's for Stock and Account Value Comparison Report in ERPNext

# Description:

### Test Summary:
  - This test suite validates the functionality of the Stock and Account Value Comparison report in ERPNext. It ensures accurate detection and handling of discrepancies between Stock Ledger Entries (SLE) and General Ledger (GL) entries under different conditions. It also verifies data correctness, proper creation of reposting entries, and validates time conversion logic used within the report.

**Report Name:- Stock and Account Value Comparison**
 - Stock Ledger Entries (SLE): Quantifies value of stock items in warehouses.
 - General Ledger Entries (GLE): Financial accounting values in relevant stock accounts

### Scenarios Covered:
  **1.Setup & Prerequisites:**
 - Creation of test data including Company, Item, Warehouse, HSN Code, Stock Entry, Stock Ledger Entry, and GL Entry.
 - Ensures test data is idempotent (doesn't create duplicates).
  
 **2.Perpetual Inventory Checks:**
 - Verifies successful execution when perpetual inventory is enabled.
 - Raises validation error when perpetual inventory is disabled.
  
  **3.Data Extraction Functions:**
 - Validates get_data, get_gl_data, get_stock_ledger_data, and get_columns functionalities.
  
  **4.Reposting Entries:**
 - Creates reposting entries when mismatch is found.
 - Validates correct handling of duplicate reposting creation.
 - Accepts both dict and JSON string inputs.
 - Confirms correct document creation in Repost Item Valuation.
  
  **5.Data Filtering and Difference Calculation:**
 - Ensures mismatches with value differences > 0.1 are captured and flagged.
 - Simulates mismatched SLE and GL entries and verifies expected logic.
  
  **6.Time Conversion Logic:**
 - Verifies conversion of posting_time from datetime.time to timedelta.
 - Handles entries with null or missing posting_time values gracefully.
  
  **7.Miscellaneous:**
 - Ensures coverage of edge cases (missing keys, null values, repeated processing).
 - Includes helper to ensure fiscal year exists and is linked to the company.

### Technology:
  1.Framework: FrappeTestCase (Unit Test Framework from Frappe)
  2.ERP System: ERPNext
  3.Language: Python 3.x

### Test Case Id's:
**T_SAAVC_001**
- Validates that report columns and data contain expected fields and correct difference calculations

**T_SAAVC_002**
- Checks that filtering by account returns data with relevant account and stock values.

**T_SAAVC_003**
- Confirms that the report raises an error when perpetual inventory is disabled for the company.

**T_SAAVC_004**
- Tests if the report detects and returns entries where the difference between stock and account values is significant.

**T_SAAVC_005**
- Verifies that the posting_time field in the report data is properly converted to a timedelta object.

**T_SAAVC_006**
- Ensures reposting entries are created correctly and the user is notified via a message.

  
### API/Report Execution
N/A

### Linked Feature/Bug
N/A

### Issue Tracker ID
N/A

### CODE REVIEW

**Comment by aasif-patel** [comment]
@ashishverma8848 please update the title and do add TS Id's in test case method and in the PR description as well
@pallavi238848

File: erpnext/stock/report/stock_and_account_value_comparison/test_stock_and_account_value_comparison.py [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+)
+
+
+class TestStockAndAccountValueComparison(FrappeTestCase):
+	def setUp(self):
+		self.company = "_Test Company"
+		self.item_code = "_Test Item"
+		self.warehouse = "_Test WareHouse"
+		self.account = "Stock In Hand - _TC"
+		self.posting_date = nowdate()
```
Comment (aasif-patel) [inline erpnext/stock/report/stock_and_account_value_comparison/test_stock_and_account_value_comparison.py]: @ashishverma8848 not creating or checking if the data exist or not

File: erpnext/stock/report/stock_and_account_value_comparison/test_stock_and_account_value_comparison.py [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+		self.assertEqual(get_sle_data[0]["posting_time"], timedelta(hours=10, minutes=15, seconds=30))
+
+		# Second entry should remain None
+		self.assertIsNone(get_sle_data[1]["posting_time"])
+
+		# Third entry should not have 'posting_time'
+		self.assertNotIn("posting_time", get_sle_data[2])
+
+
+def get_or_create_fiscal_year(company):
```
Comment (aasif-patel) [inline erpnext/stock/report/stock_and_account_value_comparison/test_stock_and_account_value_comparison.py]: we already have this method in one of the doctype test file,

**Review by aasif-patel — CHANGES_REQUESTED** [review]

File: erpnext/stock/report/stock_and_account_value_comparison/test_stock_and_account_value_comparison.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+	get_data,
+	get_gl_data,
+	get_stock_ledger_data,
+)
+
+
+class TestStockAndAccountValueComparison(FrappeTestCase):
+	def setUp(self):
+		if not frappe.db.exists("Company", "_Test Company"):
+			self.company = create_company("_Test Company")
```
Comment (aasif-patel) [inline erpnext/stock/report/stock_and_account_value_comparison/test_stock_and_account_value_comparison.py]: Create company does not return any object

File: erpnext/stock/report/stock_and_account_value_comparison/test_stock_and_account_value_comparison.py [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+		else:
+			frappe.db.set_value(
+				"Warehouse", "_Test Warehouse - _TC", "company", "_Test Company", update_modified=False
+			)
+			self.warehouse = frappe.db.get_value("Warehouse", "_Test Warehouse - _TC", "name")
+
+		if not frappe.db.exists("Item", "_Test Item"):
+			self.item_code = create_item(item_code="_Test Item", valuation_rate=100)
+		else:
+			self.item_code = "_Test Item"
```
Comment (aasif-patel) [inline erpnext/stock/report/stock_and_account_value_comparison/test_stock_and_account_value_comparison.py]: why checking item existence the method create_item from test_item already handling this

**Review by aasif-patel — CHANGES_REQUESTED** [review]
General method should be declare after the test cases

**Review by aasif-patel — APPROVED** [review]

**Comment by swathibollineni2** [comment]
@ashishverma8848  All test cases getting same issue
 NameError: Year start date or end date is overlapping with 2025. To avoid please set company
